connect/src/main/java/com/stripe/android/connect/manager/EmbeddedComponentCoordinator.kt

```diff
@@ -0,0 +1,287 @@
+package com.stripe.android.connect.manager
+
+import android.Manifest
+import android.app.Activity
+import android.app.Application.ActivityLifecycleCallbacks
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.ContextCompat.checkSelfPermission
+import com.stripe.android.connect.BuildConfig
+import com.stripe.android.connect.EmbeddedComponentManager.Configuration
+import com.stripe.android.connect.FetchClientSecretCallback
+import com.stripe.android.connect.PrivateBetaConnectSDK
+import com.stripe.android.connect.StripeEmbeddedComponent
+import com.stripe.android.connect.analytics.ComponentAnalyticsService
+import com.stripe.android.connect.analytics.ConnectAnalyticsService
+import com.stripe.android.connect.appearance.Appearance
+import com.stripe.android.connect.appearance.fonts.CustomFontSource
+import com.stripe.android.connect.di.StripeConnectComponent
+import com.stripe.android.connect.util.findActivity
+import com.stripe.android.connect.webview.ChooseFileActivityResultContract
+import com.stripe.android.connect.webview.serialization.ConnectInstanceJs
+import com.stripe.android.connect.webview.serialization.toJs
+import com.stripe.android.core.Logger
+import com.stripe.android.financialconnections.FinancialConnectionsSheet
+import com.stripe.android.financialconnections.FinancialConnectionsSheetResult
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.suspendCancellableCoroutine
+import javax.inject.Inject
+import kotlin.coroutines.resume
+
+@OptIn(PrivateBetaConnectSDK::class)
+@EmbeddedComponentManagerScope
+internal class EmbeddedComponentCoordinator @Inject constructor(
+    private val configuration: Configuration,
+    private val fetchClientSecretCallback: FetchClientSecretCallback,
+    private val logger: Logger,
+    appearance: Appearance,
+    private val customFonts: List<CustomFontSource>,
+) {
+    private val _appearanceFlow = MutableStateFlow(appearance)
```

Comment:
👀 The impl of this class is entirely moved from `EmbeddedComponentManager` without modification.